src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -80,13 +80,21 @@ public TimeSpan HeartbeatInterval
             set => this.HeartbeatProvider.HeartbeatInterval = value;
         }
 
+        /// <summary>
+        /// Gets a list of default heartbeat property providers that are disabled and will not contribute to the
+        /// default heartbeat properties.
+        /// </summary>
+        public IList<string> ExcludedHeartbeatPropertyProviders
```

Comment:
Since we already provide ExcludedHeartbeatProperties for customers to exclude any field they dont want. What additional purpose does ExcludedHeartbeatPropertyProviders serve? Can you give an example of HeartbeatPropertyProviders ?

Reply:
Yes, this is a departure from what I originally created and there is a bit of background on why I made this choice:

Before, when the `static HeartbeatDefaultPayload` class was  created, all the default properties where immediately available (or not) and there was no problems testing the associated code flows. When I added the Azure IMS default properties to the base SDK, things changed in that I had to spawn a thread and gather properties asynchronously (to cope with http requests/timeouts). This made it very difficult to test the `HeartbeatDefaultPayload`/`HeartbeatProvider.Initialize` flow as waiting for the async call to return was not exposed to anything but two methods within the `HeartbeatDefaultPayload` class.

Once I re-architected how we gather ‘default’ heartbeat properties in an async-friendly way, *in a testable* way, I saw that it was going to be a lot of work for customers to exclude every single field from any particular provider. I decided then to allow per-field exclusions *plus* per-module exclusions. Of course, this isn’t entirely necessary at the  moment since I moved the Azure IMS module over to -dotnet-server, but I figured the extra architecture doesn’t detract from anything and it allows us to easily add new default heartbeat modules in the future.

Reply:
ok. Please give an example name of a provider in the comments.